### PR TITLE
Tabular: Added load_data_internal function to TabularPredictor

### PR DESCRIPTION
**Pull request by Innixma**
*Issue #, if available:*
#249, #428, #429 

*Description of changes:*

Added load_data_internal function to TabularPredictor.

Now when users use transform_features(), they can get the y labels as well by calling these functions.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Comment by szha** [comment]
Job PR-434-1 is done. 
Docs are uploaded to http://autogluon-staging.s3-website-us-west-2.amazonaws.com/PR-434/1/index.html

**Comment by jwmueller** [comment]
Might these methods (load_X_train, load_y_train, load_X_val, load_y_val) might be better as internal methods (e.g. in Learner)?  I'm just worried if the Predictor offers too many methods, then UX may become more complicated than necessary. 

If these methods become internal, there could just be certain args in `transform_features()` to invoke each (combination of) these.  For example: `dataset` arg in `transform_features()` could have special str options `train` & `validation`.  I also don't see why `transform_features()` shouldn't automatically return a tuple with both X and y (the latter whenever it's available in data, otherwise returned `y = None`; or could even have a boolean arg `transform_label` in `transform_features()` which determines whether y is returned or not). 

Another related point that the documentation might explicitly state somewhere is:   `transform_features()` (and these new load_* methods) allow you to train other 3rd party ML models using the same data representation used internally by AutoGluon. 

If this is the main intended use-case, I'd also consider rethinking whether `transform_features()` returns the train or validation DataFrame by default (when `dataset=None`), as I think returning train might make more sense in this context (or even both train and val).  If the above suggestion is adopted with special `dataset` options `train` & `validation`, then you could also just set `dataset=train` as the default in `transform_features()`.

**Comment by Innixma** [comment]
Addressed comments based on feedback, combined functionality into a single function.

**Comment by szha** [comment]
Job PR-434-2 is done. 
Docs are uploaded to http://autogluon-staging.s3-website-us-west-2.amazonaws.com/PR-434/2/index.html

File: autogluon/task/tabular_prediction/predictor.py [hunk 1]
```diff
@@ -361,10 +361,14 @@ def transform_features(self, dataset=None, model=None, base_models=None):
                 For non-bagged mode predictors:
                     The dataset used when not specified is the validation set.
                     This can either be an automatically generated validation set or the user-defined `tuning_data` if passed during fit().
+                    If all parameters are unspecified, then the output is equivalent to `predictor.load_data_internal(dataset='val', return_X=True, return_y=False)[0]`.
```
Comment (jwmueller) [inline autogluon/task/tabular_prediction/predictor.py:364]: why is the validation set the default? I think I'm missing what the use-case would be there.   I thought the main use case would be for users to train 3rd party models on the same data representation, in which case the training set would be the default?

File: autogluon/task/tabular_prediction/predictor.py [hunk 2]
```diff
@@ -547,6 +551,56 @@ def save(self):
         self._learner.save()
         logger.log(20, "TabularPredictor saved. To load, use: TabularPredictor.load(\"%s\")" % self.output_directory)
 
+    def load_data_internal(self, dataset='train', return_X=True, return_y=True):
+        """
+        Loads the generic internal data representation used during model training.
```
Comment (jwmueller) [inline autogluon/task/tabular_prediction/predictor.py:556]: unsure what 'generic' means here?  could be confusing for users

File: autogluon/task/tabular_prediction/predictor.py (hunk 1 above)
Comment (Innixma) [reply autogluon/task/tabular_prediction/predictor.py:364]: It must default to ‘val’ for non-bagged because it would crash if it was ‘train’ and the user asked for weighted ensemble input (because it doesn’t exist)

File: autogluon/task/tabular_prediction/predictor.py (hunk 2 above)
Comment (Innixma) [reply autogluon/task/tabular_prediction/predictor.py]: Good catch, removed.
